<?xml version="1.0"?>

<robot name="quadrotor" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:property name="M_PI" value="3.1415926535897931" />
    <xacro:include filename="$(find intelligent_drone)/urdf/quadrotor_base.urdf.xacro" />
    <xacro:quadrotor_base_macro />

    <xacro:include filename="$(find intelligent_drone)/sensors/hokuyo_utm30lx.urdf.xacro" />
    <xacro:hokuyo_utm30lx name="laser0" parent="base_link" ros_topic="scan" update_rate="40"
                          ray_count="18" min_angle="-90" max_angle="90">
      <origin xyz="0.0 0.0 -0.097" rpy="${M_PI} 0 0"/>
    </xacro:hokuyo_utm30lx>

    <xacro:include filename="$(find intelligent_drone)/sensors/generic_camera.urdf.xacro" />

    <xacro:generic_camera name="front_cam" parent="base_link" ros_topic="front_cam/image"
                          cam_info_topic="front_cam/camera_info" update_rate="10" res_x="1280" res_y="1080"
                          image_format="R8G8B8" hfov="90">
      <origin xyz="0.4 0 0 " rpy="0 1.57 0"/>
    </xacro:generic_camera>


    <xacro:include filename="$(find hector_quadrotor_gazebo)/urdf/quadrotor_plugins.gazebo.xacro" />

</robot>
